app/controllers/application_controller.rb

```diff
@@ -1,10 +1,166 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do
+    erb :index
+  end
+
+
+  get '/users/:id' do 
+    @user = User.find_by(id: params[:id])
+    erb :'users/show'
+  end
+
+  get '/signup' do
+   if !session[:user_id]
```

Comment:
You can use your `logged_in?` helper method here, which is preferable to referring to the session object directly.